tardis/visualization/tools/tests/test_sdec_plot.py

```diff
@@ -0,0 +1,295 @@
+"""Tests for SDEC Plots."""
+from tardis.base import run_tardis
+from tardis.io.config_reader import Configuration
+import pytest
+import pandas as pd
+import numpy as np
+from copy import deepcopy
+from tardis.visualization.tools.sdec_plot import SDECData, SDECPlotter
+import astropy.units as u
+import h5py
+
+
+@pytest.fixture(scope="module")
+def config_verysimple(atomic_dataset):
+    config = Configuration.from_yaml(
+        "tardis/io/tests/data/tardis_configv1_verysimple.yml"
+    )
+    return config
+
+
+@pytest.fixture(scope="module")
+def simulation_simple(config_verysimple, atomic_dataset):
+    """Instantiate SDEC plotter using a simple simulation model."""
+    # Setup simulation configuration using config_verysimple and
+    # override properties in such a way to make the simulation run faster
+    config_verysimple.montecarlo.iterations = 3
+    config_verysimple.montecarlo.no_of_packets = 4000
+    config_verysimple.montecarlo.last_no_of_packets = -1
+    config_verysimple.spectrum.virtual.virtual_packet_logging = True
+    config_verysimple.spectrum.num = 2000
+
+    atomic_data = deepcopy(atomic_dataset)  # TODO: why deepcopy?
+    sim = run_tardis(
+        config_verysimple,
+        atom_data=atomic_data,
+        show_convergence_plots=False,
+    )
+    return sim
+
+
+class TestSDECPlotter(object):
+    @classmethod
+    @pytest.fixture(scope="class", autouse=True)
+    def setup(self, request):
+        cls = type(self)
+        cls.hdf_file = h5py.File("sdec_ref.h5", "a")
+        cls.save = False
+        cls.ref_data_path = ""
+        if request.config.getoption("--generate-reference"):
+            cls.save = True
+
+    @pytest.fixture(scope="class")
+    def plotter(self, simulation_simple):
+        return SDECPlotter.from_simulation(simulation_simple)
+
+    @pytest.mark.parametrize("species", [["Si II", "Ca II", "C", "Fe I-V"]])
+    def test_parse_species_list(self, request, plotter, species):
+        plotter._parse_species_list(species)
+        subgroup_name = request.node.callspec.id
+        if self.save:
+            group = self.hdf_file.create_group(subgroup_name)
+            group.create_dataset(
+                "_full_species_list", data=plotter._full_species_list
+            )
+            group.create_dataset("_species_list", data=plotter._species_list)
+            group.create_dataset("_keep_colour", data=plotter._keep_colour)
```

Comment:
I'm not sure why you're saving data in test functions? 

The data should already be saved in ref-data or locally and then you should read it from there when comparing against what tests are producing

Reply:
I'm not actually- when tests are run, the saved data in a file called `sdec_ref.h5` is used for testing. That extra code is used to generate reference data only when the user asks for it, using the `--generate-reference` option when running tests. 
I will soon create a PR on the tardis-refdata repository for this too!